plonk-napi/src/proof.rs

```diff
@@ -40,22 +41,491 @@ macro_rules! impl_proof {
+                        ft_eval1,
+                        public: public_,
+                        prev_challenges_scalars: prev_challenges_scalars.0,
+                        prev_challenges_comms,
+                    }
+                }
+
+                #[napi(getter)]
+                pub fn commitments(&self) -> NapiProverCommitments {
+                    self.commitments.clone()
+                }
+                #[napi(getter)]
+                pub fn proof(&self) -> NapiOpeningProof {
+                    self.proof.clone()
+                }
+                #[napi(getter)]
+                pub fn evals(&self) -> NapiProofEvaluations {
+                    self.evals.clone()
+                }
+                #[napi(getter, js_name="public_")]
+                pub fn public_(&self) -> NapiFlatVector<$NapiF> {
+                    self.public.clone()
+                }
+                #[napi(getter, js_name="prev_challenges_scalars")]
+                pub fn prev_challenges_scalars(&self) -> NapiVecVecF {
+                    [<NapiVecVec $field_name:camel>](self.prev_challenges_scalars.clone())
+                }
+                #[napi(getter, js_name="prev_challenges_comms")]
+                pub fn prev_challenges_comms(&self) -> NapiVector<$NapiPolyComm> {
+                    self.prev_challenges_comms.clone()
+                }
+
+                #[napi(setter, js_name="set_commitments")]
+                pub fn set_commitments(&mut self, commitments: NapiProverCommitments) {
+                    self.commitments = commitments
+                }
+                #[napi(setter, js_name="set_proof")]
+                pub fn set_proof(&mut self, proof: NapiOpeningProof) {
+                    self.proof = proof
+                }
+                #[napi(setter, js_name="set_evals")]
+                pub fn set_evals(&mut self, evals: NapiProofEvaluations) {
+                    self.evals = evals
+                }
+                #[napi(setter, js_name="set_public_")]
+                pub fn set_public_(&mut self, public_: NapiFlatVector<$NapiF>) {
+                    self.public = public_
+                }
+                #[napi(setter, js_name="set_prev_challenges_scalars")]
+                pub fn set_prev_challenges_scalars(&mut self, prev_challenges_scalars: NapiVecVecF) {
+                    self.prev_challenges_scalars = prev_challenges_scalars.0
+                }
+                #[napi(setter, js_name="set_prev_challenges_comms")]
+                pub fn set_prev_challenges_comms(&mut self, prev_challenges_comms: NapiVector<$NapiPolyComm>) {
+                    self.prev_challenges_comms = prev_challenges_comms
+                }
+
+                #[napi]
+                #[allow(deprecated)]
+                pub fn serialize(&self) -> String {
+                    let (proof, _public_input) = self.into();
+                    let serialized = rmp_serde::to_vec(&proof).unwrap();
```

Comment:
Reminds me of the Nori bug https://github.com/o1-labs/proof-systems/pull/3369/, let's keep an eye open with these ones 👀